controllers/front/submitplus.php

```diff
@@ -0,0 +1,266 @@
+                    'id_invoice' => 0,
+                    'shipping' => 0,
+                    'currency' => $payment->transactions[0]->amount->currency,
+                    'payment_date' => date("Y-m-d H:i:s"),
+                );
+
+                if ($submit == 'confirmPayment') {
+
+                    if ($payment->state == 'approved') {
+
+                        $paypal->validateOrder(
+                            $this->id_cart, $this->getOrderStatus('payment'), $payment->transactions[0]->amount->total,
+                            $payment->payer->payment_method, null, $transaction
+                        );
+                        $return['success'][] = 'Votre paiement a bien été pris en compte.';
+                    } else {
+
+                        $paypal->validateOrder(
+                            $this->id_cart, $this->getOrderStatus('payment_error'),
+                            $payment->transactions[0]->amount->total, $payment->payer->payment_method, null,
+                            $transaction
+                        );
+                        $return['error'][] = 'Une erreur est survenue pendant le paiement.';
+                    }
+                } elseif ($submit == 'confirmCancel') {
+
+                    $paypal->validateOrder(
+                        $this->id_cart, $this->getOrderStatus('order_canceled'),
+                        $payment->transactions[0]->amount->total, $payment->payer->payment_method, null, $transaction
+                    );
+                    $return['success'][] = 'Votre commande a bien été annulé.';
+                } else
+                    $return['error'][] = 'Une erreur est survenue pendant le paiement.';
+            } else
+                $return['error'][] = 'Une erreur est survenue pendant le paiement.';
+        } else
+            $return['error'][] = 'Une erreur est survenue pendant le paiement.';
+
+        echo json_encode($return);
+        die();
+    }
+
+    /**
+     * Execute the hook displayPaymentReturn
+     */
+    public function displayPaymentReturn()
+    {
+        $params = $this->displayHook();
+
+        if ($params && is_array($params)) return Hook::exec('displayPaymentReturn', $params, (int) $this->id_module);
+        return false;
+    }
+
+    /**
+     * Execute the hook displayOrderConfirmation
+     */
+    public function displayOrderConfirmation()
+    {
+        $params = $this->displayHook();
+
+        if ($params && is_array($params)) return Hook::exec('displayOrderConfirmation', $params);
+        return false;
+    }
+
+    public function getOrderStatus($template)
+    {
+        /*
+         * payment
+         * payment_error
+         * order_canceled
+         * refund
+         */
+        return Db::getInstance()->getValue('SELECT id_order_state FROM '._DB_PREFIX_.'order_state_lang WHERE template = "'.$template.'" AND id_lang = "'.$this->context->language->id.'"');
```

Comment:
Please sanitize the values of this request, with an `(int)` for interger values and `pSQL(...)` for strings